### DISCUSSION TITLE
What should I do to keep my progress running?

### DISCUSSION THREAD

**Discussion by Foves5** [cc-tweaked/CC-Tweaked#1191] [Q&A] [answered]
I wrote a program like this: 
```
while (1) do
    sleep(1)
    redstone.setAnalogOutput('top', 15)
    sleep(1)
    redstone.setAnalogOutput('top', 0)
end
```
It seems will run forever, but when I enter the world again, I found that the process stopped.
What should I do if I want the program keep running when I renter the world?

**Comment by Lupus590** [comment, marked as answer]
You need to make a startup file.

The easiest way is to make a file in the root of the CC computer's filesystem called `startup` (edit may add .lua to this, that's fine), in this startup you then need to `shell.run` your current code. Alternatively, instead of shell.run you could copy your code into the startup file.

**Reply by Foves5** [reply]
Very Thanks. : )
